xds/src/main/java/io/grpc/xds/LoadStatsStoreImpl.java

```diff
@@ -125,27 +122,25 @@ public ClusterStats generateLoadReport() {
 
   @Override
   public void addLocality(final Locality locality) {
-    ClientLoadCounter counter = localityLoadCounters.get(locality);
-    checkState(counter == null || !counter.isActive(),
-        "An active counter for locality %s already exists", locality);
+    ReferenceCounted<ClientLoadCounter> counter = localityLoadCounters.get(locality);
     if (counter == null) {
-      localityLoadCounters.put(locality, new ClientLoadCounter());
-    } else {
-      counter.setActive(true);
+      counter = ReferenceCounted.wrap(new ClientLoadCounter());
+      localityLoadCounters.put(locality, counter);
     }
+    counter.retain();
   }
 
   @Override
   public void removeLocality(final Locality locality) {
-    ClientLoadCounter counter = localityLoadCounters.get(locality);
-    checkState(counter != null && counter.isActive(),
-        "No active counter for locality %s exists", locality);
-    counter.setActive(false);
+    ReferenceCounted<ClientLoadCounter> counter = localityLoadCounters.get(locality);
+    counter.release();
   }
 
+  @Nullable
   @Override
   public ClientLoadCounter getLocalityCounter(final Locality locality) {
-    return localityLoadCounters.get(locality);
+    ReferenceCounted<ClientLoadCounter> counter = localityLoadCounters.get(locality);
+    return counter == null || counter.getReferenceCount() == 0 ? null : counter.get();
```

Comment:
Why changed this?

Reply:
Following the same specification of the API as before, if the locality is is not under track, return `null`. We don't want to break in the middle as that makes this API fragile.